### DISCUSSION TITLE
Turborepo custom remote cache configuration

### DISCUSSION THREAD

**Discussion by AndreiCBogdan** [vercel/turborepo#5182] [Help] [answered]
Hi there, we're trying to setup a remote caching using turbo ` v1.5.5`[ turborepo-remote-cache](https://github.com/ducktors/turborepo-remote-cache) with AWS S3 storage and can see the following error in the log relating to the line of code below when running the turbo cli remote command from a github ci runner. When running locally the request goes through without an error.

`{"severity":"WARNING","level":40,"time":1684164667390,"pid":7,"hostname":"turborepo-cache-57d6c996db-m8ktx","reqId":"WmEucNymQiefEGn9CLNAYw-0","data":null,"isBoom":true,"isServer":false,"output":{"statusCode":400,"payload":{"statusCode":400,"error":"Bad Request","message":"Missing Authorization header"},"headers":{}},"stack":"Error: Missing Authorization header\n    at Object.<anonymous> (/home/app/node/src/plugins/remote-cache/index.ts:56:27)\n    at hookIterator (/home/app/node/node_modules/.pnpm/fastify@4.12.0/node_modules/fastify/lib/hooks.js:246:10)\n    at next (/home/app/node/node_modules/.pnpm/fastify@4.12.0/node_modules/fastify/lib/hooks.js:174:16)\n    at hookRunner (/home/app/node/node_modules/.pnpm/fastify@4.12.0/node_modules/fastify/lib/hooks.js:196:3)\n    at Object.routeHandler [as handler] (/home/app/node/node_modules/.pnpm/fastify@4.12.0/node_modules/fastify/lib/route.js:469:7)\n    at Router.callHandler (/home/app/node/node_modules/.pnpm/find-my-way@7.6.0/node_modules/find-my-way/index.js:398:14)\n    at Router.lookup (/home/app/node/node_modules/.pnpm/find-my-way@7.6.0/node_modules/find-my-way/index.js:376:17)\n    at Server.preRouting (/home/app/node/node_modules/.pnpm/fastify@4.12.0/node_modules/fastify/fastify.js:773:14)\n    at Server.emit (node:events:513:28)\n    at Server.emit (node:domain:489:12)","type":"Error","message":"Missing Authorization header"}`

The following env vars are set locally and on the ci runner

TURBO_TOKEN: XXXXX
TURBO_TEAM: team
TURBO_API: XXXX
TURBO_REMOTE_ONLY: true

When running:
`turbo run build`
locally the turbo cli sends posts requests to the turborepo remote server to cache the artifacts in s3 but on the gihub ci runner the error above is shown in the server logs and the cache artifacts don't make it to s3.
Won't lie, not too sure what aspect of this is unhappy but expect it's something to do with our config :) any help would be appreciated please!

**Comment by mehulkar** [comment, marked as answer]
The stack trace you posted shows `fastify`, which is a dependency of the [`turborepo-remote-cache`](https://github.com/ducktors/turborepo-remote-cache/blob/eb9411c648e1a2d2e595b9017bf4504b30b6ef98/package.json#L57) implementation you're using. This may be a better issue to file in that repo.
